### PR TITLE
[RFR] Playable game

### PR DESCRIPTION
**Pull request by HALLERPierre**
[Trello](https://trello.com/c/wVZtGqha/2-2-as-fred-i-want-to-be-able-to-play)

# Todo
- [x] Copy/paste re-usable code from react native app
- [x] Add docker network to allow communication between isomorphic app & api (waiting for deploy)
- [x] Create a new game from API
- [x] Assign me a team
- [x] Move cube

# Next PR
- [ ] Create a solo game
- [ ] Fix tests

### CODE REVIEW

File: src/components/Instructions.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import styled from 'styled-components';
+import { CIRCLE_VALUE, NEUTRAL_VALUE } from '../constants/game';
+
+const CIRCLE_SRC = '/static/circle.png';
+const CROSS_SRC = '/static/cross.png';
+
+const Instructions = ({ team, isPlaying, winner }) => {
```
Comment (Kmaschta) [inline src/components/Instructions.js:10]: Don't forget the prop-types for your components

Comment (HALLERPierre) [reply src/components/Instructions.js:10]: Done

File: src/components/Cube.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import styled from 'styled-components';
+
+import { CIRCLE_VALUE, CROSS_VALUE } from '../constants/game';
+
+const CIRCLE_SRC = '/static/circle.png';
+const CROSS_SRC = '/static/cross.png';
+const NEUTRAL_SRC = 'static/neutral.png';
+
+const getLogo = value => {
```
Comment (zyhou) [inline src/components/Cube.js]: You can write something like :

```js
const getLogo = value => ({
    CIRCLE_VALUE: '/static/circle.png',
    CROSS_VALUE: '/static/cross.png'
}[value]) || 'static/neutral.png';
```

File: src/pages/game.js [hunk 3]
```diff
@@ -1,36 +1,89 @@
...
+
+const registerHooks = (game, dispatch) => {
+    const { id, myTeam, currentPlayer } = game;
+    const isPlaying = isMyTurn(myTeam, currentPlayer);
+
+    useEffect(refreshGame(id, isPlaying, dispatch), null);
+    useEffect(fetchMovables(id, dispatch), [currentPlayer]);
+};
+
+const Game = ({ initGame, myTeam }) => {
```
Comment (zyhou) [inline src/pages/game.js:28]: Proptypes

File: src/components/Cube.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+    const handlePress = isMovable ? handlePressCube : () => {};
+    return (
+        <StyledCube isMovable={isMovable} isWinning={isWinning} onClick={handlePress}>
+            {image}
+        </StyledCube>
+    );
+};
+
+Cube.propTypes = {
```
Comment (zyhou) [inline src/components/Cube.js:33]: I think `isMovable`, `handlePressCube`, `isWinning` isn't required props. Don't forget `defaultProps`

File: src/components/Instructions.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+            <TeamInscrution>
+                <InstructionText>{"You're"} playing with </InstructionText>
+                <Logo src={logoSrc} />
+            </TeamInscrution>
+            <InstructionText>{isPlaying ? 'Your turn !' : 'Waiting for the opponent...'}</InstructionText>
+        </InstructionContainer>
+    );
+};
+
+Instructions.propTypes = {
```
Comment (zyhou) [inline src/components/Instructions.js:40]: default props

**Review by zyhou — COMMENTED** [review]
Don't hesite to split in multiple PR (don't forget test :) )

File: src/components/Cube.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        [CROSS_VALUE]: CROSS_SRC,
+    }[value] || NEUTRAL_SRC);
+
+const Cube = props => {
+    const testId = getTestId(props);
+    const { isMovable, handlePressCube, isSelected, isWinning, value } = props;
+    const logo = getLogo(value);
+    const image = !isSelected ? <CubeImage src={logo} /> : null;
+
+    const handlePress = isMovable ? handlePressCube : () => {};
```
Comment (Kmaschta) [inline src/components/Cube.js]: Instead of creating an empty function, `<StyledCube>` should accept `null` and do nothing IMO
This is duck typing

**Comment by Kmaschta** [comment]
We'll just merge #5

**Pull request closed without merging**